core/src/main/java/org/infinispan/util/concurrent/locks/LockContainer.java

```diff
@@ -0,0 +1,51 @@
+package org.infinispan.util.concurrent.locks;
+
+import org.infinispan.util.concurrent.locks.impl.InfinispanLock;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A container for locks
+ *
+ * @author Manik Surtani
+ * @author Mircea.Markus@jboss.com
+ * @author Pedro Ruivo
+ * @since 4.0
+ */
+public interface LockContainer {
+
+   /**
+    * @param key the key to lock.
+    * @return the lock for a specific object to be acquired. If the lock does not exists, it is created.
+    */
+   ExtendedLockPromise acquire(Object key, Object lockOwner, long time, TimeUnit timeUnit);
+
+   /**
+    * @param key the key to lock.
+    * @return the lock for a specific object. If the lock does not exists, it return {@code null}.
+    */
+   InfinispanLock getLock(Object key);
```

Comment:
I don't think it's a good idea to return an impl class from a non-impl interface method. Maybe we can remove this method, and replace it with `getOwner(key)`?